service/src/main/java/uk/gov/hmcts/reform/fpl/service/email/content/AbstractEmailContentProvider.java

```diff
@@ -40,6 +50,19 @@ protected AbstractEmailContentProvider(String uiBaseUrl) {
             .put("caseUrl", uiBaseUrl + "/case/" + JURISDICTION + "/" + CASE_TYPE + "/" + caseDetails.getId());
     }
 
+    protected ImmutableMap.Builder<String, Object> getSDOPersonalisationBuilder(CaseDetails caseDetails, CaseData caseData) {
+        return ImmutableMap.<String, Object>builder()
+            .put("familyManCaseNumber", Objects.isNull(caseData.getFamilyManCaseNumber()) ? "" : caseData.getFamilyManCaseNumber() + ",")
```

Comment:
I've imported the static method, i didn't use defaultIfNull here though because i wanted to be consistent and i felt that defaultIfNull would work for family man, but not for the others which call another method inside isNull and then gives null pointer exception, what do you think?

Reply:
```
protected ImmutableMap.Builder<String, Object> getSDOPersonalisationBuilder(CaseDetails caseDetails, CaseData caseData) {
    ImmutableMap.Builder builder = ImmutableMap.builder();

    builder.put("familyManCaseNumber", defaultIfNull(caseData.getFamilyManCaseNumber(), "");
    if (isNotNull(caseData.getRespondents1)) {
        builder.put("leadRespondentsName", StringUtils.capitalize(caseData.getRespondents1().get(0).getValue().getParty().getLastName()));
    }

    builder.put("hearingDate", defaultIfNull(hearingBookingService.getMostUrgentHearingBooking(caseData.getHearingDetails()), ""));
    builder.put("reference", String.valueOf(caseDetails.getId()));
    builder.put("caseUrl", uiBaseUrl + "/case/" + JURISDICTION + "/" + CASE_TYPE + "/" + caseDetails.getId());

    return builder;
  }
```

Reply:
thoughts ☝️

Reply:
the hearingDate might need another if outside. I don't know what your thoughts on it are but inline code like this I find very hard to read 

```
.put("hearingDate",Objects.isNull(hearingBookingService.getMostUrgentHearingBooking(caseData.getHearingDetails())) ? "" : dateFormatterService.formatLocalDateToString(hearingBookingService.getMostUrgentHearingBooking(caseData.getHearingDetails()).getDate(),FormatStyle.LONG))
```

Reply:
hmm it is hard to read but the above didn't work for me as 1) if there is no respondents and it doesn't put anything into the builder for respondents like an empty string it throws an exception and 2) I was concatenating a comma on the end of respondent and family man if they are present so i can't use defaultIfNull when i want to do that because it never returns a null value i.e the comma is always there?

Reply:
respondent issue, I guess an else for the if statement where leadRespondent, empty string. This was why I was asking what happens if you don't send any data for a placeholder in the notify templates

your comma issue: defaultIfNull(thingYouWant + comma, the ThingForNull)